variables.tf

```diff
@@ -1,9 +1,81 @@
-########################################################################################################################
-# Input Variables
-########################################################################################################################
-
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "infra_type" {
+  type        = string
+  description = "Type of target agent infrastructure. Allowed values: `ibm_kubernetes`, `ibm_openshift` and `ibm_satellite`."
+  default     = "ibm_kubernetes"
+  validation {
+    condition     = contains(["ibm_kubernetes", "ibm_openshift", "ibm_satellite"], var.infra_type)
```

Comment:
I thought you said openshift wasn't support yet for agent?

Reply:
Yes, not yet but will be coming in soon. See this docs for available infra_type values https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/schematics_agent#infra_type